### DISCUSSION TITLE
Telemetry and personal data

### DISCUSSION THREAD

**Discussion by serkor1** [posit-dev/positron#5160] [Q&A] [answered]
Hi!

Thank you for the amazing work. I was wondering how much telemetry and other data Positron (being a fork of VS Code) collects? According to this [blog post](https://www.roboleary.net/tools/2022/04/20/vscode-telemetry.html) and this [gist](https://gist.github.com/robole/97f6e1c4dc888ae0a49a7683e54949c2) VS Code collects thousands of data points, and even the developers at VSCodium says they can't control it "all". They are, however, active suppressing as much collection of data as possible.

What is the stance of Positron on this matter? And how much telemetry and other data is gonna be collected by Positron, if any, relative to R Studio?

Best,

Serkan

**Comment by juliasilge** [comment, marked as answer]
Thanks for this question, @serkor1! Currently, we [have taken a similar approach as VSCodium](https://github.com/posit-dev/positron/pull/2882) at turning off as much telemetry as is possible. On a personal level, I agree with that blog post that the telemetry in VS Code is excessive but we as a team do expect to have legitimate (in our eyes/opinion, at least) purposes for collecting some data eventually. We will prioritize transparency with users about any changes/additions to telemetry and you will be able to see, for example, PRs making such changes. 

To give you an idea of what we are considering, we currently plan to: 

- collect data on OS usage (Windows vs. Mac vs. Linux) via clicks to download an installer, once we have a non-GH place to get installers (this would not be in-product telemetry, but from events on the website)
- collect data on whether folks are using R and/or Python (this _would_ be in-product telemetry)

**Reply by serkor1** [reply]
This was a comforting read @juliasilge, thank you. I also noticed that telemetry is switched off by default in Positron - so thank you for that, too.

How about telemetry at the extension level? It is my understanding that the Telemetry in VS Code is being collected via the editor, and separately via extensions. The latter being independent of the editor settings. Is there a possibility and/or a plan to restrict extensions from collecting data? If its even possible at all that is.

Best,

**Reply by juliasilge** [reply]
We don't currently have plans to change the behavior of extensions, and honestly I am not sure how feasible that would be. When it comes to privacy concerns around extensions, using Positron is about equivalent to using VSCodium in that we both use [Open VSX](https://open-vsx.org/) rather than the Microsoft marketplace but do not take any special actions about what the extensions themselves do.
